Submitted By: Andrew Benton <andy@benton.eu.com>
Date: 2012-03-23
Initial Package Version: 0.48.2
Upstream Status: unknown
Origin: Fedora
http://lists.fedoraproject.org/pipermail/scm-commits/2012-March/751440.html
Description: Fixes compiling inkscape-0.48.2 with gcc-4.7.0

--- inkscape-0.48.2/src/graphlayout.cpp.orig	2011-07-08 19:24:45.468790000 +0100
+++ inkscape-0.48.2/src/graphlayout.cpp	2012-03-23 21:42:59.956641322 +0000
@@ -155,11 +155,12 @@
          ++i)
     {
         SPItem *iu=*i;
-        map<string,unsigned>::iterator i=nodelookup.find(iu->getId());
-        if(i==nodelookup.end()) {
+        map<string,unsigned>::iterator i_iter=nodelookup.find(iu->getId());
+        map<string,unsigned>::iterator i_iter_end=nodelookup.end();
+        if(i_iter==i_iter_end) {
             continue;
         }
-        unsigned u=i->second;
+        unsigned u=i_iter->second;
         GSList *nlist=iu->avoidRef->getAttachedConnectors(Avoid::runningFrom);
         list<SPItem *> connectors;
 
